var/spack/repos/builtin/packages/xsdk/package.py

```diff
@@ -111,8 +149,26 @@ class Xsdk(BundlePackage):
     # these are type='build' dependencies, but spack reports a conflict anyway.
     # This will be fixed once the new concretizer becomes available
     # (says @adamjstewart)
-    depends_on('phist@develop kernel_lib=tpetra ~fortran ~scamac ~openmp ~host', when='@develop +phist')
-    depends_on('phist@1.7.5 kernel_lib=tpetra ~fortran ~scamac ~openmp ~host', when='@0.4.0 +phist')
+
+    depends_on('phist kernel_lib=tpetra', when='+trilinos +phist')
+    depends_on('phist kernel_lib=petsc', when='~trilinos +phist')
+    depends_on('phist@develop ~fortran ~scamac ~openmp ~host', when='@develop +phist')
+    depends_on('phist@1.8.0 ~fortran ~scamac ~openmp ~host', when='@0.5.0 +phist')
+    depends_on('phist@1.7.5 ~fortran ~scamac ~openmp ~host', when='@0.4.0 +phist')
+
+    depends_on('ginkgo@develop ~openmp', when='@develop +ginkgo')
+    depends_on('ginkgo@develop ~openmp+cuda', when='@develop +ginkgo +cuda')
+    depends_on('ginkgo@1.1.0 ~openmp', when='@0.5.0 +ginkgo')
+    depends_on('ginkgo@1.1.0 ~openmp+cuda', when='@0.5.0 +cuda +ginkgo')
```

Comment:
Why not use the same OpenMP treatment above?

Reply:
At some point we have add openmp variant to xsdk and consistently enable/disable it for all packages.

Then there is this issue of different openmp versions that we need to figure out [some packages need openmp version 4.5+]. So we haven't worked out this yet. Does spack has some infrastructure for detecting and using this info in pacakges?

```
diff --git a/var/spack/repos/builtin/packages/xsdk/package.py b/var/spack/repos/builtin/packages/xsdk/package.py
index 354db12d6..c13f33100 100644
--- a/var/spack/repos/builtin/packages/xsdk/package.py
+++ b/var/spack/repos/builtin/packages/xsdk/package.py
@@ -156,10 +153,9 @@ class Xsdk(BundlePackage):
     depends_on('phist@1.8.0 ~fortran ~scamac ~openmp ~host', when='@0.5.0 +phist')
     depends_on('phist@1.7.5 ~fortran ~scamac ~openmp ~host', when='@0.4.0 +phist')
 
-    depends_on('ginkgo@develop ~openmp', when='@develop +ginkgo')
-    depends_on('ginkgo@develop ~openmp+cuda', when='@develop +ginkgo +cuda')
-    depends_on('ginkgo@1.1.0 ~openmp', when='@0.5.0 +ginkgo')
-    depends_on('ginkgo@1.1.0 ~openmp+cuda', when='@0.5.0 +cuda +ginkgo')
+    depends_on('ginkgo +cuda', when='+ginkgo +cuda')
+    depends_on('ginkgo@develop', when='@develop +ginkgo')
+    depends_on('ginkgo@1.1.0', when='@0.5.0 +ginkgo')
 
     depends_on('py-libensemble@develop+petsc4py', when='@develop +libensemble')
     depends_on('py-libensemble@0.5.2+petsc4py', when='@0.5.0 +libensemble')

```

```
diff --git a/var/spack/repos/builtin/packages/ginkgo/package.py b/var/spack/repos/builtin/packages/ginkgo/package.py
index a0de875a5..06affc022 100644
--- a/var/spack/repos/builtin/packages/ginkgo/package.py
+++ b/var/spack/repos/builtin/packages/ginkgo/package.py
@@ -23,7 +23,7 @@ class Ginkgo(CMakePackage, CudaPackage):
 
     variant('shared', default=True, description='Build shared libraries')
     variant('full_optimizations', default=False, description='Compile with all optimizations')
-    variant('openmp', default=sys.platform != 'darwin',  description='Build with OpenMP')
+    variant('openmp', default=True,  description='Build with OpenMP')
     variant('develtools', default=False, description='Compile with develtools enabled')
     variant('build_type', default='Release',
             description='The build type to build',
@@ -31,6 +31,7 @@ class Ginkgo(CMakePackage, CudaPackage):
 
     depends_on('cmake@3.9:', type='build')
     depends_on('cuda@9:',    when='+cuda')
+    depends_on('llvm-openmp', when='+openmp %clang platform=darwin')
 
     conflicts('%gcc@:5.2.9')

```
I could change the above to enable openmp [similar to tasmanian] -  would have to redo all testing

I get same failure on OSX+clang. Perhaps I'm doing something wrong here..


```
==> Executing phase: 'cmake'
==> [2019-11-11-14:06:57.630744] 'cmake' '/var/folders/xk/v0fd809s2y7b3x8bp05l2nww0000gq/T/balay/spack-stage/spack-stage-ginkgo-1.1.0-b6ggjffdhcohmns4v2soomf65ipykpjd/spack-src' '-G' 'Unix Makefiles' '-DCMAKE_INSTALL_PREFIX:PATH=/Users/balay/spack/opt/spack/darwin-catalina-x86_64/clang-11.0.0-apple/ginkgo-1.1.0-b6ggjffdhcohmns4v2soomf65ipykpjd' '-DCMAKE_BUILD_TYPE:STRING=Release' '-DCMAKE_VERBOSE_MAKEFILE:BOOL=ON' '-DCMAKE_FIND_FRAMEWORK:STRING=LAST' '-DCMAKE_FIND_APPBUNDLE:STRING=LAST' '-DCMAKE_INSTALL_RPATH_USE_LINK_PATH:BOOL=FALSE' '-DCMAKE_INSTALL_RPATH:STRING=/Users/balay/spack/opt/spack/darwin-catalina-x86_64/clang-11.0.0-apple/ginkgo-1.1.0-b6ggjffdhcohmns4v2soomf65ipykpjd/lib;/Users/balay/spack/opt/spack/darwin-catalina-x86_64/clang-11.0.0-apple/ginkgo-1.1.0-b6ggjffdhcohmns4v2soomf65ipykpjd/lib64;/Users/balay/spack/opt/spack/darwin-catalina-x86_64/clang-11.0.0-apple/llvm-openmp-9.0.0-kkckpcgbwofd7nmpry7nq2bdgq3gvdec/lib' '-DCMAKE_PREFIX_PATH:STRING=/Users/balay/spack/opt/spack/darwin-catalina-x86_64/clang-11.0.0-apple/llvm-openmp-9.0.0-kkckpcgbwofd7nmpry7nq2bdgq3gvdec' '-DGINKGO_BUILD_CUDA=OFF' '-DGINKGO_BUILD_OMP=ON' '-DBUILD_SHARED_LIBS=ON' '-DGINKGO_JACOBI_FULL_OPTIMIZATIONS=OFF' '-DGINKGO_DEVEL_TOOLS=OFF' '-DGINKGO_BUILD_BENCHMARKS=OFF' '-DGINKGO_BUILD_DOC=OFF' '-DGINKGO_BUILD_EXAMPLES=OFF' '-DGINKGO_BUILD_TESTS=OFF'
-- The C compiler identification is AppleClang 11.0.0.11000033
-- The CXX compiler identification is AppleClang 11.0.0.11000033
-- Check for working C compiler: /Users/balay/spack/lib/spack/env/clang/clang
-- Check for working C compiler: /Users/balay/spack/lib/spack/env/clang/clang -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /Users/balay/spack/lib/spack/env/clang/clang++
-- Check for working CXX compiler: /Users/balay/spack/lib/spack/env/clang/clang++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Performing Test Ginkgo_C_COVERAGE_SUPPORTED
-- Performing Test Ginkgo_C_COVERAGE_SUPPORTED - Success
-- Performing Test Ginkgo_C_TSAN_SUPPORTED
-- Performing Test Ginkgo_C_TSAN_SUPPORTED - Failed
-- Performing Test Ginkgo_C_ASAN_SUPPORTED
-- Performing Test Ginkgo_C_ASAN_SUPPORTED - Success
-- Performing Test Ginkgo_CXX_COVERAGE_SUPPORTED
-- Performing Test Ginkgo_CXX_COVERAGE_SUPPORTED - Success
-- Performing Test Ginkgo_CXX_TSAN_SUPPORTED
-- Performing Test Ginkgo_CXX_TSAN_SUPPORTED - Failed
-- Performing Test Ginkgo_CXX_ASAN_SUPPORTED
-- Performing Test Ginkgo_CXX_ASAN_SUPPORTED - Success
-- Looking for C++ include cxxabi.h
-- Looking for C++ include cxxabi.h - found
-- Could NOT find PAPI (missing: PAPI_LIBRARY PAPI_INCLUDE_DIR) 
CMake Error at /usr/local/Cellar/cmake/3.15.5/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:137 (message):
  Could NOT find OpenMP_C (missing: OpenMP_C_FLAGS OpenMP_C_LIB_NAMES)
Call Stack (most recent call first):
  /usr/local/Cellar/cmake/3.15.5/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:378 (_FPHSA_FAILURE_MESSAGE)
  /usr/local/Cellar/cmake/3.15.5/share/cmake/Modules/FindOpenMP.cmake:477 (find_package_handle_standard_args)
  omp/CMakeLists.txt:1 (find_package)


-- Configuring incomplete, errors occurred!
```

cc: @hartwiganzt

Reply:
Currently Spack doesn't have any info on which versions of OpenMP a particular compiler supports. We have a hacky way of detecting C++?? support, but not OpenMP version. Ideally there would be a way to require a compiler with both.